### DISCUSSION TITLE
Where is the rule designer for LibreClinica?

### DISCUSSION THREAD

**Discussion by shanzhu2021** [reliatec-gmbh/LibreClinica#222] [Q&A] [answered]
I deployed the libreclinica and created a new study. For the task 'Create Rules', I have two actions - 'Add' and 'Rule Designer'. When I click 'Add', I first download a template XML file from the 'Template' and try to to import rule data by upload the XML file, but it report error (see the screenshot), and when I click 'Rule Designer', the link broken.

<img width="480" alt="rules1" src="https://user-images.githubusercontent.com/77312250/108332672-093afe80-720b-11eb-855b-6246675554a9.PNG">

So what is the best practice I should follow for rules?

**Comment by toskrip** [comment, marked as answer]
RuleDesigner is a separate application. We do not maintain or distribute it. You can find the source code of RuleDesigner in its own repository here:

https://github.com/OpenClinica/Designer

But currently the RuleDesigner is not working with LibreClinica (see issue #188). We are thinking of fixing this issue but from long term perspective I would say that XML rule upload is preferred practice as there is nobody currently maintaining RuleDesigner code base. We also do not know whether it can run on new tomcat/java combo. If you are looking for learning resources regarding writing valid rules, you can find some examples under the Rules section on this web:

http://www.trialdatasolutions.com/tds/howto/index.jsp
